fhirstore/search/urlparser.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+import re
+
+# Format the url args to a dict
+def url_to_dict(url_args):
+    search_args = {key: url_args.getlist(key) for key in url_args.keys()}
+    return search_args
+
+
+# Parse commas in a (key,value)
+def parse_comma(key, value):
+    has_comma = "," in value
+    if has_comma:
+        return "multiple", {key: value.split(",")}
+    else:
+        return key, [value]
+
+
+# Process all the dict for possible commas
+def pre_process_params(url_args):
+    search_args = url_to_dict(url_args)
+    processed_params = defaultdict(list)
+    if search_args == {}:
+        processed_params = {}
+
+    for key, value in search_args.items():
+        if len(value) == 1:
```

Comment:
I'm not sure you need to have this if, because I think you're doing similar things in both case.
If you do `for element in value` and value has length 1, you'll process the element the same way